packages/x-components/src/components/animations/create-directional-animation-factory.ts

```diff
@@ -0,0 +1,32 @@
+import Vue, { VueConstructor } from 'vue';
+import { capitalize } from '../../utils/string';
+
+/**
+ * Abstract Factory to create animations Factory. The returned animation factory uses the
+ * `animationName` parameter to create a Transition Component with that name.
+ *
+ * @param animationName - The name to use in the Transition Component of animation.
+ * @returns The animation factory configured.
+ *
+ * @internal
+ */
+export function createDirectionalAnimationFactory(
+  animationName: string
+): (animationOrigin?: AnimationOrigin) => VueConstructor {
+  return (animationOrigin = 'top') => {
+    return Vue.extend({
+      name: `Transition${capitalize(animationName)}${capitalize(animationOrigin)}`,
```

Comment:
If you want to leave the name in kebab case it is fine, now its kinda a monster mixing kebab, pascal, and camelcase🤣 but fortunately Vue devtools are able to transform it to pascal case properly.

```suggestion
      name: `${animationName}-${animationOrigin}`,
```

![image](https://user-images.githubusercontent.com/68222542/153187058-ae90de64-4018-41e8-8a3b-e2bbcac3c2fd.png)

Reply:
You're right. Fortunately, Vue devtools already transforms from `kebab-case` to `PascalCase`.

Note: if you want to keep the `Transition` prefix, don't commit Javi's suggestion directly and do the change manually 😅 .

Reply:
done